functions/java-jre.bash

```diff
@@ -17,60 +34,153 @@ java_webupd8_archive() {
-    if [ $? -ne 0 ]; then echo "FAILED"; rm -f ${file}; exit 1; fi
-    rm -rf $file ${jdkInstallLocation:?}/*
-    mv ${jdkTempLocation}/* ${jdkInstallLocation}/; rmdir ${jdkTempLocation}
 
-    javaPath=$(echo $downloadPath|sed 's|https://cdn.azul.com/zulu-embedded/bin/||')
-    javaPath=$(echo $javaPath|sed 's|.tar.gz||')
-    cond_redirect update-alternatives --remove-all java
-    cond_redirect update-alternatives --remove-all javac
-    cond_redirect update-alternatives --install /usr/bin/java java $jdkInstallLocation/$javaPath/bin/java 1083000
-    cond_redirect update-alternatives --install /usr/bin/javac javac $jdkInstallLocation/$javaPath/bin/javac 1083000
-    echo $jdkInstallLocation/$javaPath/lib/$archName > /etc/ld.so.conf.d/java.conf
-    echo $jdkInstallLocation/$javaPath/lib/$archName/jli >> /etc/ld.so.conf.d/java.conf
-    cond_redirect ldconfig
-    if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; exit 1; fi
+    if is_x86_64; then 
+      dpkg --add-architecture i386
+      cond_redirect apt update
+      cond_redirect apt -y install libc6:i386 libncurses5:i386 libstdc++6:i386
+    fi
 
-  else
-    echo -n "$(timestamp) [openHABian] Installing Zulu Enterprise OpenJDK... "
-    cond_redirect apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 219BD9C9
-    if [ $? -ne 0 ]; then echo "FAILED (keyserver)"; exit 1; fi
-    if is_ubuntu; then
-      echo "deb $arch http://repos.azulsystems.com/ubuntu stable main" > /etc/apt/sources.list.d/zulu-enterprise.list
-    else
-      echo "deb $arch http://repos.azulsystems.com/debian stable main" > /etc/apt/sources.list.d/zulu-enterprise.list
+  elif [ "$1" == "64-bit" ]; then
+    echo -n "$(timestamp) [openHABian] Installing Java Zulu 64-Bit OpenJDK... "
+    if is_arm; then 
+      link="$(fetch_zulu_tar_url "arm-64-bit")";
+      jdkArch="aarch64"
+    else 
+      link="$(fetch_zulu_tar_url "x86-64-bit")";
+      jdkArch="amd64"
     fi
-    cond_redirect apt-get update
-    cond_redirect apt-get -y install zulu-8
-    if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; exit 1; fi
+
+  else 
+    echo -n "[DEBUG] Unvalid argument to function java_zulu_8_tar()"
+    exit 1
   fi
-  if [ -f /usr/lib/systemd/system/openhab2.service ]; then
-    cond_redirect systemctl start openhab2.service
+  jdkTempLocation="$(mktemp -d /tmp/.XXXXXXXXXX)"
+  jdkInstallLocation="/opt/jdk"
+  mkdir -p $jdkInstallLocation
+
+  # Fetch and copy new JAVA runtime
+  cond_redirect wget -nv -O "$jdkTempLocation"/zulu8.tar.gz "$link"
+  tar -xpzf "$jdkTempLocation"/zulu8.tar.gz -C "${jdkTempLocation}"
+  if [ $? -ne 0 ]; then echo "FAILED"; rm -rf "$jdkTempLocation"/zulu8.tar.gz; exit 1; fi
+  rm -rf "$jdkTempLocation"/zulu8.tar.gz "${jdkInstallLocation:?}"/*
+  mv "${jdkTempLocation}"/* "${jdkInstallLocation}"/
+
+  # Install Cryptography Extension Kit to enable cryptos using more then 128 bits
+  jdkSecurity=$(find "${jdkInstallLocation}"/*/lib/security ... -print -quit)
```

Comment:
* can only match on dir, on ARM */jre/lib/security works.
But sure that opt/jdk contains only one match? Older zulu installs in this folder will make it fail (only first one ist matched).

Reply:
I rewrote this code to:
```
jdkPath="$(readlink -f $(which java))"
jdkSecurity="$(dirname ${jdkPath})/../lib/security"
```
I think that one will be safe.

Reply:
Ist this before update-alternatives? (Sorry watching on my mobile now). If yes maybe  better to use folder name from temp dir....

Reply:
No is invoked after `update-alternatives` so new installation will be reflected in the `which java` call.